src/main/java/net/neoforged/neoforge/transfer/ResourceHandlerUtil.java

```
@@ -0,0 +1,23 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.transfer;
+
+import net.neoforged.neoforge.transfer.resources.IResource;
+
+public final class ResourceHandlerUtil {
+    private ResourceHandlerUtil() {}
+
+    /**
+     * Determines if either the given resource or amount is classified as empty.
+     *
+     * @param resource The resource to check.
+     * @param amount   An amount to check. <strong>Must be non-negative.</strong>
```

Do we want to allow it to be non-negative and just return true? It might be more useful as a util method to have that behavior. At the very least if we don't then we probably should have it throw on invalid values. Otherwise it isn't really "must"

---

Personally I would like to throw to ensure anytime a negative value is used throughout the resource and handling packages, it properly stops you. The concern was "`isEmpty` doesn't give the implication it should throw", despite being documented to do so. 

We can change this to do `<= 0`, but again, I'd much rather have us enforce negative values not be present at all rather than hope for proper use.

---

I don't really have that much of a preference, just so long as if we aren't having it handle negative numbers, we throw instead of letting it potentially affect things further down the stream.